test/terraform/mzcompose.py

```diff
@@ -534,3 +537,422 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
             )
             run_ignore_error(["kubectl", "delete", "namespace", "materialize"])
             spawn.runv(["terraform", "destroy", "-auto-approve"], cwd=path)
+
+
+def workflow_gcp(c: Composition, parser: WorkflowArgumentParser) -> None:
+    """To run locally use `aws sso login` first."""
+    parser.add_argument(
+        "--setup",
+        default=True,
+        action=argparse.BooleanOptionalAction,
+        help="Run setup steps",
+    )
+    parser.add_argument(
+        "--cleanup",
+        default=True,
+        action=argparse.BooleanOptionalAction,
+        help="Destroy the region at the end of the workflow.",
+    )
+    parser.add_argument(
+        "--tag",
+        type=str,
+        help="Custom version tag to use",
+    )
+    parser.add_argument(
+        "files",
+        nargs="*",
+        default=COMPATIBLE_TESTDRIVE_FILES,
+        help="run against the specified files",
+    )
+
+    args = parser.parse_args()
+
+    tag = args.tag or f"v{ci_util.get_mz_version()}--pr.g{git.rev_parse('HEAD')}"
+    materialize_environment = None
+    environmentd_port_forward_process = None
+    balancerd_port_forward_process = None
+
+    path = MZ_ROOT / "test" / "terraform" / "gcp"
+
+    gcp_service_account_json = os.getenv("GCP_SERVICE_ACCOUNT_JSON")
+    assert (
+        gcp_service_account_json
+    ), "GCP_SERVICE_ACCOUNT_JSON environment variable has to be set"
+    gcloud_creds_path = path / "gcp.json"
+    with open(gcloud_creds_path, "w") as f:
+        f.write(gcp_service_account_json)
+    os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = str(gcloud_creds_path)
+
+    try:
+        spawn.runv(["gcloud", "config", "set", "project", "materialize-ci"])
+
+        spawn.runv(
+            [
+                "gcloud",
+                "auth",
+                "activate-service-account",
+                f"--key-file={gcloud_creds_path}",
+            ],
+        )
+
+        spawn.runv(["gcloud", "services", "enable", "container.googleapis.com"])
+        spawn.runv(["gcloud", "services", "enable", "sqladmin.googleapis.com"])
+        spawn.runv(
+            ["gcloud", "services", "enable", "cloudresourcemanager.googleapis.com"]
+        )
+        spawn.runv(["gcloud", "services", "enable", "servicenetworking.googleapis.com"])
+        spawn.runv(["gcloud", "services", "enable", "iamcredentials.googleapis.com"])
```

Comment:
I think that once you run those commands, the services are enabled, but it probably doesn't hurt to run them every time in case the service account ever changes.